src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -165,6 +170,68 @@ private static void setModules(TestInputConfiguration.Builder testInputConfigBui
             throw new CheckstyleException("Config not specified on top."
                 + "Please see other inputs for examples of what is required.");
         }
+
+        final boolean isXmlConfig = "/*xml".equals(lines.get(0));
+        if (isXmlConfig) {
+            final String stringXmlConfig = getXmlConfig(lines);
+            final InputSource inputSource = new InputSource(new StringReader(stringXmlConfig));
+            final Configuration xmlConfig = ConfigurationLoader.loadConfiguration(
+                inputSource, new PropertiesExpander(System.getProperties()),
+                    ConfigurationLoader.IgnoredModulesOptions.EXECUTE
+            );
+            final String configName = xmlConfig.getName();
+            if (!"Checker".equals(configName)) {
+                throw new CheckstyleException(
+                        "First module should be Checker, but was " + configName);
+            }
+            handleXmlConfig(testInputConfigBuilder, inputFilePath, xmlConfig.getChildren());
+        }
+        else {
+            handleKeyValueConfig(testInputConfigBuilder, inputFilePath, lines);
+        }
+    }
+
+    private static String getXmlConfig(List<String> lines) {
+        final String dtdConfiguration = "<?xml version=\"1.0\"?>\n"
+            + "<!DOCTYPE module PUBLIC\n"
+            + "        \"-//Checkstyle//DTD Checkstyle Configuration 1.3//EN\"\n"
+            + "        \"https://checkstyle.org/dtds/configuration_1_3.dtd\">";
+        final StringBuilder xmlConfig = new StringBuilder(dtdConfiguration);
+        for (String line : lines) {
+            if (line.startsWith("/*xml")) {
+                continue;
+            }
+            if (line.startsWith("*/")) {
+                break;
+            }
+            xmlConfig.append(line);
```

Comment:
Please make a single method for both models that load as set of lines content from start line to finish line.
So content with xml or properties is simply big block of lines 
After that we just need loading of xml config by standard methods that we have from InputStream .